src/Nethermind/Nethermind.Blockchain/Spec/ChainHeadSpecProvider.cs

```diff
@@ -62,7 +63,9 @@ public IReleaseSpec GetCurrentHeadSpec()
             lock (_lock)
             {
                 _lastHeader = headerNumber;
-                return _headerSpec = GetSpec(headerNumber);
+                if (header is not null)
+                    return _headerSpec = _specProvider.GetSpec(header);
+                return _headerSpec = GetSpec((ForkActivation)headerNumber);
```

Comment:
wait, this is incorrect

Reply:
uses the whole BlockHeader if header is not null
line 67